models/netsuite2/intermediate/int_netsuite2__acctxperiod_exchange_rate_map.sql

```diff
@@ -22,27 +22,40 @@ consolidated_exchange_rates as (
     from {{ var('netsuite2_consolidated_exchange_rates') }}
 ),
 
+currencies as (
+    select *
+    from {{ var('netsuite2_currencies') }}
+),
+
 period_exchange_rate_map as ( -- exchange rates used, by accounting period, to convert to parent subsidiary
   select
     consolidated_exchange_rates.accounting_period_id,
+    consolidated_exchange_rates.accounting_book_id,
     consolidated_exchange_rates.average_rate,
     consolidated_exchange_rates.current_rate,
     consolidated_exchange_rates.historical_rate,
     consolidated_exchange_rates.from_subsidiary_id,
-    consolidated_exchange_rates.to_subsidiary_id
+    consolidated_exchange_rates.to_subsidiary_id,
+    to_subsidiaries.name as to_subsidiary_name,
+    currencies.symbol as to_subsidiary_currency_symbol
   from consolidated_exchange_rates
+  
+  left join subsidiaries to_subsidiaries
+    on consolidated_exchange_rates.to_subsidiary_id = to_subsidiaries.subsidiary_id
+
+  left join currencies
+    on currencies.currency_id = to_subsidiaries.currency
 
-  where consolidated_exchange_rates.to_subsidiary_id in (select subsidiary_id from subsidiaries where parent_id is null)  -- constraint - only the primary subsidiary has no parent
```

Comment:
This constraint anchors downstream models to the root parent (i.e. `parent_id is null`). By removing it and including the `to_subsidiary` in subsequent joins, models contain the full range of consolidated transactions available in NetSuite reporting.

This is important when reporting is needed at different levels of consolidation. See https://github.com/fivetran/dbt_netsuite/issues/93